araproc/analysis/daq_quality_cut.py

```diff
@@ -0,0 +1,253 @@
+import numpy as np
+import logging
+import yaml
+from araproc.framework import constants
+import importlib.resources as pkg_resources
+import araproc.framework.config_files as config_files
+
+def process_event_info(useful_event, station_id):
+
+    """
+    Process a useful_event to extract event information, determine trigger type, and calculate block length.
+
+    Parameters
+    ----------
+    useful_event : object
+        The useful event object containing event information.
+    station_id : int
+        The station ID used to determine pulser time.
+
+    Returns
+    -------
+    blk_len : np.ndarray
+        Calculated block length for the event.
+    trig_type : int
+        Trigger type (0: RF trigger, 1: Pulser trigger, 2: Soft trigger).
+    block_nums : np.ndarray
+        Block numbers for the event.
+    channel_mask : np.ndarray
+        Channel mask for the event.
+    """
+
+    # Extract block information
+    num_blocks = len(useful_event.blockVec)
+    block_nums = np.empty(num_blocks)  # Create an empty array for block numbers
+    channel_mask = np.empty(num_blocks, dtype=int)  # Create an empty array for channel masks
+    for n in range(num_blocks):
+        this_block_id = useful_event.blockVec[n].getBlock()
+        block_nums[n] = int(this_block_id)
+        channel_mask[n] = useful_event.blockVec[n].channelMask
+    # Extract event-level information
+    read_win = useful_event.numReadoutBlocks  # Number of readout blocks
+
+    if useful_event.isRFTrigger():
```

Comment:
I can't remember if calpulsers are also rf triggers on ARA. They are on PA in which case the order here matters. @clark2668 do you know off the top of your head?

Reply:
Yup! That is correct. Cal pulsers _are_ RF triggers. They can be "tagged" based on their timing under normal operating conditions. But they fundamentally are RF triggers.

Reply:
Hmm. So, the option would be something like this, right or I should consider sth else as well? 
![image](https://github.com/user-attachments/assets/669260b1-17de-4af9-8012-0f6d56971055)

Reply:
Ya you can do it like that or like:
```
if useful_event.isCalpulserEvent():
  trig_type = 1
elif useful_event.isRFTrigger():
  trig_type = 0
```